### ISSUE TITLE
Align streaming with regular generation

### ISSUE THREAD

**Issue by RobinPicard** [dottxt-ai/outlines#1482]
Currently models can be called. That creates a `Generator` that then invokes the generate `method`. Alternatively, users can create the `Generator` themselves and then call the generator. What matters is that the user do not have to call the `generate` method of the model themselves. That way, we can handle the creation of the logits proecessor for them in the `Generator`. The same should exist for streaming.
